### ISSUE TITLE
[Bug]: logs getting flooded with "java.time.DateTimeException: Missing network time fix"

### ISSUE THREAD

**Issue by techouse** [braze-inc/braze-android-sdk#28]
### Braze Android SDK Version

30.3.0

### Steps To Reproduce

Run Flutter Android app using the Flutter [`braze_plugin` v9.0.0](https://pub.dev/packages/braze_plugin/changelog#900) which uses the Braze Android SDK `v30.3.0` under the hood.

### Expected Behavior

No errors in the logs.

### Actual Incorrect Behavior

The app works fine, however, ever since upgrading from `v29.0.1` to `v30.3.0` my logs get flooded with errors like these
```
E/Braze v30.3.0 .DateTimeUtils(10408): Error reading date and time from network. Using system time instead.
E/Braze v30.3.0 .DateTimeUtils(10408): java.time.DateTimeException: Missing network time fix
E/Braze v30.3.0 .DateTimeUtils(10408): 	at java.lang.reflect.Constructor.newInstance0(Native Method)
E/Braze v30.3.0 .DateTimeUtils(10408): 	at java.lang.reflect.Constructor.newInstance(Constructor.java:343)
E/Braze v30.3.0 .DateTimeUtils(10408): 	at android.os.ParcelableException.readFromParcel(ParcelableException.java:56)
E/Braze v30.3.0 .DateTimeUtils(10408): 	at android.os.ParcelableException$1.createFromParcel(ParcelableException.java:82)
E/Braze v30.3.0 .DateTimeUtils(10408): 	at android.os.ParcelableException$1.createFromParcel(ParcelableException.java:79)
E/Braze v30.3.0 .DateTimeUtils(10408): 	at android.os.Parcel.readParcelableInternal(Parcel.java:5015)
E/Braze v30.3.0 .DateTimeUtils(10408): 	at android.os.Parcel.readParcelable(Parcel.java:4997)
E/Braze v30.3.0 .DateTimeUtils(10408): 	at android.os.Parcel.createExceptionOrNull(Parcel.java:3177)
E/Braze v30.3.0 .DateTimeUtils(10408): 	at android.os.Parcel.createException(Parcel.java:3166)
E/Braze v30.3.0 .DateTimeUtils(10408): 	at android.os.Parcel.readException(Parcel.java:3149)
E/Braze v30.3.0 .DateTimeUtils(10408): 	at android.os.Parcel.readException(Parcel.java:3091)
E/Braze v30.3.0 .DateTimeUtils(10408): 	at android.app.timedetector.ITimeDetectorService$Stub$Proxy.latestNetworkTime(ITimeDetectorService.java:462)
E/Braze v30.3.0 .DateTimeUtils(10408): 	at android.os.SystemClock.currentNetworkTimeMillis(SystemClock.java:345)
E/Braze v30.3.0 .DateTimeUtils(10408): 	at android.os.SystemClock$3.millis(SystemClock.java:392)
E/Braze v30.3.0 .DateTimeUtils(10408): 	at com.braze.support.DateTimeUtils.nowInMilliseconds(SourceFile:4)
E/Braze v30.3.0 .DateTimeUtils(10408): 	at bo.app.c5.a(SourceFile:19)
E/Braze v30.3.0 .DateTimeUtils(10408): 	at bo.app.c5$g.invokeSuspend(SourceFile:4)
E/Braze v30.3.0 .DateTimeUtils(10408): 	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
E/Braze v30.3.0 .DateTimeUtils(10408): 	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:104)
E/Braze v30.3.0 .DateTimeUtils(10408): 	at kotlinx.coroutines.internal.LimitedDispatcher$Worker.run(LimitedDispatcher.kt:111)
E/Braze v30.3.0 .DateTimeUtils(10408): 	at kotlinx.coroutines.scheduling.TaskImpl.run(Tasks.kt:99)
E/Braze v30.3.0 .DateTimeUtils(10408): 	at kotlinx.coroutines.scheduling.CoroutineScheduler.runSafely(CoroutineScheduler.kt:585)
E/Braze v30.3.0 .DateTimeUtils(10408): 	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.executeTask(CoroutineScheduler.kt:802)
E/Braze v30.3.0 .DateTimeUtils(10408): 	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.runWorker(CoroutineScheduler.kt:706)
E/Braze v30.3.0 .DateTimeUtils(10408): 	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.run(CoroutineScheduler.kt:693)
```

Using `v29.0.1` I only got 1 of these on app startup, now they are just constant and won't go away unless I set
```kotlin
BrazeLogger.logLevel = Log.ASSERT
```

### Verbose Logs

```shell
E/Braze v30.3.0 .DateTimeUtils(10408): Error reading date and time from network. Using system time instead.
E/Braze v30.3.0 .DateTimeUtils(10408): java.time.DateTimeException: Missing network time fix
E/Braze v30.3.0 .DateTimeUtils(10408): 	at java.lang.reflect.Constructor.newInstance0(Native Method)
E/Braze v30.3.0 .DateTimeUtils(10408): 	at java.lang.reflect.Constructor.newInstance(Constructor.java:343)
E/Braze v30.3.0 .DateTimeUtils(10408): 	at android.os.ParcelableException.readFromParcel(ParcelableException.java:56)
E/Braze v30.3.0 .DateTimeUtils(10408): 	at android.os.ParcelableException$1.createFromParcel(ParcelableException.java:82)
E/Braze v30.3.0 .DateTimeUtils(10408): 	at android.os.ParcelableException$1.createFromParcel(ParcelableException.java:79)
E/Braze v30.3.0 .DateTimeUtils(10408): 	at android.os.Parcel.readParcelableInternal(Parcel.java:5015)
E/Braze v30.3.0 .DateTimeUtils(10408): 	at android.os.Parcel.readParcelable(Parcel.java:4997)
E/Braze v30.3.0 .DateTimeUtils(10408): 	at android.os.Parcel.createExceptionOrNull(Parcel.java:3177)
E/Braze v30.3.0 .DateTimeUtils(10408): 	at android.os.Parcel.createException(Parcel.java:3166)
E/Braze v30.3.0 .DateTimeUtils(10408): 	at android.os.Parcel.readException(Parcel.java:3149)
E/Braze v30.3.0 .DateTimeUtils(10408): 	at android.os.Parcel.readException(Parcel.java:3091)
E/Braze v30.3.0 .DateTimeUtils(10408): 	at android.app.timedetector.ITimeDetectorService$Stub$Proxy.latestNetworkTime(ITimeDetectorService.java:462)
E/Braze v30.3.0 .DateTimeUtils(10408): 	at android.os.SystemClock.currentNetworkTimeMillis(SystemClock.java:345)
E/Braze v30.3.0 .DateTimeUtils(10408): 	at android.os.SystemClock$3.millis(SystemClock.java:392)
E/Braze v30.3.0 .DateTimeUtils(10408): 	at com.braze.support.DateTimeUtils.nowInMilliseconds(SourceFile:4)
E/Braze v30.3.0 .DateTimeUtils(10408): 	at bo.app.c5.a(SourceFile:19)
E/Braze v30.3.0 .DateTimeUtils(10408): 	at bo.app.c5$g.invokeSuspend(SourceFile:4)
E/Braze v30.3.0 .DateTimeUtils(10408): 	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
E/Braze v30.3.0 .DateTimeUtils(10408): 	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:104)
E/Braze v30.3.0 .DateTimeUtils(10408): 	at kotlinx.coroutines.internal.LimitedDispatcher$Worker.run(LimitedDispatcher.kt:111)
E/Braze v30.3.0 .DateTimeUtils(10408): 	at kotlinx.coroutines.scheduling.TaskImpl.run(Tasks.kt:99)
E/Braze v30.3.0 .DateTimeUtils(10408): 	at kotlinx.coroutines.scheduling.CoroutineScheduler.runSafely(CoroutineScheduler.kt:585)
E/Braze v30.3.0 .DateTimeUtils(10408): 	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.executeTask(CoroutineScheduler.kt:802)
E/Braze v30.3.0 .DateTimeUtils(10408): 	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.runWorker(CoroutineScheduler.kt:706)
E/Braze v30.3.0 .DateTimeUtils(10408): 	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.run(CoroutineScheduler.kt:693)
```

### Additional Information

Device: Google Pixel 6
OS: Android 14 (SDK 34)

**Comment by radixdev** [comment]
Hi @techouse ,

This was addressed and fixed in version 30.4.0 of the SDK. Please let us know if this resolves your issue.

julian

**Comment by techouse** [comment]
Thanks for the prompt fix @radixdev ♥️

I'll test once 30.4.0 becomes available on Maven.
